cc/dual_net.cc

```diff
@@ -33,6 +34,50 @@ using tensorflow::error::CANCELLED;
 
 namespace minigo {
 
+constexpr int DualNet::kNumStoneFeatures;
+constexpr int DualNet::kNumBoardFeatures;
+
+void DualNet::InitializeFeatures(const Position& position,
+                                 BoardFeatures* features) {
+  const auto my_color = position.to_play();
+  const auto their_color = OtherColor(my_color);
+  const float to_play = my_color == Color::kBlack ? 1 : 0;
+
+  for (int i = 0; i < kN * kN; ++i) {
+    int j = i * kNumStoneFeatures;
+    auto stone_color = position.stones()[i].color();
+    auto my_stone = stone_color == my_color ? 1 : 0;
+    auto their_stone = stone_color == their_color ? 1 : 0;
+    for (int plane = 0; plane < kPlayerFeature; plane += 2) {
+      (*features)[j++] = my_stone;
+      (*features)[j++] = their_stone;
+    }
+    (*features)[j++] = to_play;
+  }
+}
+
+void DualNet::UpdateFeatures(const BoardFeatures& old_features,
```

Comment:
Someone later will be happier if there's a comment on this.  It's quite Clever(tm)

Reply:
Done.